migration/migrator/migrations/system/20220417174632_patch_authentication_method.py

```diff
@@ -0,0 +1,29 @@
+"""Migration for the Submitty system."""
+from collections import OrderedDict
+import json
+
+def up(config):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    """
+    authentication_file = config.config_path / 'authentication.json'
```

Comment:
Should this be 
```python
authentication_file = config.config_path + '/authentication.json' 
```

instead?

Reply:
I'm not really sure to be honest. I had just copied that line from the original migration that created the authentication file as shown: https://github.com/Submitty/Submitty/blob/d78f916ce2419dae839f2a041d7887c6cb2d0c19/migration/migrator/migrations/system/20211214211432_authentication_method.py#L17

I'm not too familiar with Python so I wasn't sure if this was some idiomatic method of creating the path but I saw it worked.

Reply:
`config.config_path` is an instance of [`pathlib.Path`](https://docs.python.org/3/library/pathlib.html). It defines an [overload of the slash operator](https://docs.python.org/3/library/pathlib.html#operators) to make it easier to do path joining. Various equivalent methods for this are:

```python
from pathlib import Path

a = Path('foo')

b = a / 'bar' / 'baz'
c = Path(a, 'bar', 'baz')
d = Path.joinpath('bar', 'baz')

assert b == c == d  # foo/bar/baz
```

Trying to do `config.config_path + '/authentication.json'` will throw error: `TypeError: unsupported operand type(s) for +: 'PosixPath' and 'str'`.